cookbooks/omnibus-supermarket/libraries/config.rb

```diff
@@ -1,58 +1,223 @@
-require 'fileutils'
-require 'securerandom'
+require 'mixlib/config'
+require 'chef/mixin/deep_merge'
 
-# Manages configuration
-class Supermarket
+module Supermarket
```

Comment:
So, did we decide to change the config file format? The reason it was done the way it was done in the first place is that if we do it like this, we have to define and document everything in one or two places: the attributes file and here.

This also makes it so that if you add a configurable value in the app, you must add it here as well, making some more maintenance pain.

I'm cool with having the conversation around completely changing how we configure this, and I'm totally open to it, but it seems like it's out of scope for this PR. Is there anything in making a combined mode that requires us to change this as well?